### PR TITLE
Correct generation of version number in OS X

### PR DESCRIPTION
**Pull request by KaiZin2018**
Derive version.h from version.wx also in Mac OS build with Xcode.
Generate the name of DMG-packet in Mac OS from version.wx.

This patch will be the last to close issue #605.

### CODE REVIEW

File: Makefile.macos [hunk 1]
```diff
@@ -32,3 +32,17 @@ dist:
 # Local variables:
 # mode: makefile
 # End:
+ifndef VER_SPECIAL
+VER_SPECIAL = OSX
+endif
+
+src/ui/wxWidgets/version.h: version.wx Misc/mkversion.pl src/ui/wxWidgets/version.in
+ifdef VER_REF
```
Comment (ronys) [inline Makefile.macos:40]: Typo - should be VER_REV

**Review by ronys — CHANGES_REQUESTED** [review]

File: Makefile.macos (hunk 1 above)
Comment (KaiZin2018) [reply Makefile.macos]: Thank you for the correction.

**Review by ronys — APPROVED** [review]

**Comment by ronys** [comment]
Mac build is broken:
   /bin/sh -c /Users/runner/Library/Developer/Xcode/DerivedData/pwsafe-xcode6-botpwyrcakxdikczrovdtjytfhdg/Build/Intermediates.noindex/pwsafe-xcode6.build/Release/pwsafe.build/Script-E6CFEEF711EA3E2F001402D8.sh
Usage: Misc/mkversion.pl MAJOR=x MINOR=y [REV=z] [SPECIAL=w] template outfile
make: *** [src/ui/wxWidgets/version.h] Error 1
Command PhaseScriptExecution failed with a nonzero exit code

**Comment by KaiZin2018** [comment]
Pull rauest #730 will fix.

**Comment by KaiZin2018** [comment]
Pull request...

**Comment by KaiZin2018** [comment]
Sorry, it look like, that I'm also blocked to create a pull request from my last two commit. I even don't know how to perform this.

Please adapt Makefile.macos starting from line 39
39 39 src/ui/wxWidgets/version.h: version.wx Misc/mkversion.pl src/ui/wxWidgets/version.in
40 40 ifdef VER_REV
41       -	@Misc/mkversion.pl MAJOR=$(VER_MAJOR) MINOR=$(VER_MINOR) REV=$(VER_REV) \
42      -
43      -                                        SPECIAL=$(VER_SPECIAL) src/ui/wxWidgets/version.in $@
     41 +	@Misc/mkversion.pl MAJOR=$(VER_MAJOR) MINOR=$(VER_MINOR) REV=$(VER_REV) SPECIAL=$(VER_SPECIAL) src/ui/wxWidgets/version.in $@
44 42 else
45      -	@Misc/mkversion.pl MAJOR=$(VER_MAJOR) MINOR=$(VER_MINOR) \
46      -                                        SPECIAL=$(VER_SPECIAL) src/ui/wxWidgets/version.in $@
47      -
     43 +	@Misc/mkversion.pl MAJOR=$(VER_MAJOR) MINOR=$(VER_MINOR) SPECIAL=$(VER_SPECIAL) src/ui/wxWidgets/version.in $@
48 44 endif

The adaptation of README.MAC.DEVELOPERS.md is not so important and can be done later.

**Comment by KaiZin2018** [comment]
Looks like that I cannot format well, I should use the preview

**Comment by KaiZin2018** [comment]
39 39 src/ui/wxWidgets/version.h: version.wx Misc/mkversion.pl src/ui/wxWidgets/version.in
40 40 ifdef VER_REV
41 ...- @Misc/mkversion.pl MAJOR=$(VER_MAJOR) MINOR=$(VER_MINOR) REV=$(VER_REV) 
42 ...-
43 ...- SPECIAL=$(VER_SPECIAL) src/ui/wxWidgets/version.in $@
... 41 + @Misc/mkversion.pl MAJOR=$(VER_MAJOR) MINOR=$(VER_MINOR) REV=$(VER_REV) SPECIAL=$(VER_SPECIAL) src/ui/wxWidgets/version.in $@
44 42 else
45 ...- @Misc/mkversion.pl MAJOR=$(VER_MAJOR) MINOR=$(VER_MINOR) 
46 ...- SPECIAL=$(VER_SPECIAL) src/ui/wxWidgets/version.in $@
47 ...-
... 43 + @Misc/mkversion.pl MAJOR=$(VER_MAJOR) MINOR=$(VER_MINOR) SPECIAL=$(VER_SPECIAL) src/ui/wxWidgets/version.in $@
48 44 endif

Line 43 of new version is located at one line.

**Pull request merged**